src/paginator.ts

```diff
@@ -184,13 +183,22 @@ export class Paginator {
     const autoPaginate = parsedArguments.autoPaginate;
 
     if (autoPaginate) {
+      const results = new Array<any>();
       paginator.runAsStream_(parsedArguments, originalMethod)
         .on('error', callback)
-        .pipe(
-          concat((results: any) => {
-            callback(null, results);
-          })
-        );
+        .on('data', data => results.push(data))
+        .on('end', () => {
+          if (results.length > 0) {
```

Comment:
This section of code makes me want `concat-stream` back :) Whenever I see this pattern in user-land, I always want to ping the developer and say "Hey, y'know, there's this module called `concat-stream` that'll do that for ya."

I don't think we should flatten the dependencies too much, if it means re-inventing the wheel, thus defeating the utility of modularity that npm is good at.